storage/gcsemu/gcsemu.go

```diff
@@ -122,6 +126,18 @@ func (g *GcsEmu) Handler(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	if r.Header.Get("content-encoding") == "gzip" {
+		gzipReader, err := gzip.NewReader(r.Body)
+		if err != nil && io.EOF != err {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		if err == nil {
+			r.Body = gzipReader
+		}
+	}
+
```

Comment:
I thought Go would tend to handle this transparently, but I'm guessing you ran you a specific failure case?  You've vetted this behavior against real GCS?

Reply:
yes the real gcs was working fine in that case but the emulator was not. dont actually remember the exact case now though

Reply:
This is a good candidate to prove out with a new test case in gcsemu_test!